R/description.R

```diff
@@ -97,15 +99,20 @@ build_description_list <- function(fields = list()) {
   compact(utils::modifyList(defaults, fields))
 }
 
-check_package_name <- function(name) {
+check_package_name <- function(name, stop_for_name) {
   if (!valid_package_name(name)) {
-    ui_stop(c(
-      "{ui_value(name)} is not a valid package name. It should:",
+    msg <- c(
+      "{ui_value(name)} is not a valid package name. To be allowed on CRAN, it should:",
       "* Contain only ASCII letters, numbers, and '.'",
       "* Have at least two characters",
       "* Start with a letter",
       "* Not end with '.'"
-    ))
+    )
+    if (stop_for_name) {
+      ui_stop(msg)
+    } else {
+      ui_warn(stop_for_name)
+    }
   }
```

Comment:
I lean towards only checking when `stop_for_name = TRUE`. If people go out of their way to call `stop_for_name = FALSE`, let's assume they know what they're doing.

This makes this function simpler and the tests.

Reply:
Current behavior gives a warning because the default is `use_description(stop_for_name = FALSE)`, but making it TRUE (and called `check_name`) will probably reduce surprise.